### PR TITLE
enhancement: additional key codes

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

It will allow to use more key codes, ex. `@keydown.home=""`

**Other information:**

### CODE REVIEW

**Comment by MartinMuzatko** [comment]
Although you can add them via `Vue.config`, these are very helpful and often used keycodes. I would like to see them as default!

File: src/compiler/codegen/events.js [hunk 1]
```diff
@@ -13,6 +13,13 @@ const keyCodes: { [key: string]: number | Array<number> } = {
   left: 37,
   right: 39,
   down: 40,
+  end: 35,
+  home: 36,
+  del: 46,
+  backspace: 8,
```
Comment (jkzing) [inline src/compiler/codegen/events.js]: `backspace` is already there in 'delete'😅

Comment (jacekkarczmarczyk) [reply src/compiler/codegen/events.js]: Yes, but if you do `@keydown.delete` you don't know whether it was `del` or `backspace` pressed, i'm don't know why it was done like this

Comment (jkzing) [reply src/compiler/codegen/events.js]: Actually, you can already get this done by `@keydown.backspace`, `@keydown.page-up`, documented here: [https://vuejs.org/v2/guide/events.html#Automatic-Key-Modifers](https://vuejs.org/v2/guide/events.html#Automatic-Key-Modifers)

Comment (jacekkarczmarczyk) [reply src/compiler/codegen/events.js]: Lovely! Didn't know about this

Comment (jacekkarczmarczyk) [reply src/compiler/codegen/events.js]: Edit: so are these modifiers still required in the code? Some of them are shortcuts (like `esc` for `escape`), but `enter` or `tab` could be removed

**Pull request closed without merging**